cdap-master/src/main/java/co/cask/cdap/master/startup/ConfigurationCheck.java

```diff
@@ -191,8 +191,8 @@ private boolean isValidKafkaTopic(String key) {
 
   private boolean isValidPartitionKey(String key) {
     try {
-      LogPartitionType.valueOf(key);
-    } catch (IllegalArgumentException e) {
+      LogPartitionType.valueOf(key.toUpperCase());
+    } catch (IllegalArgumentException | NullPointerException e) {
```

Comment:
So @chtyim, you are right. Other methods in the same class could also throw unexpected exceptions. 

In my opinion, we can assume in most places in the code that this will not happen. However, since this is the startup check that is especially purposed to find issues in the configuration, it should be more resilient to missing properties. Perhaps the right thing is to change all the validation methods to catch any Exception. Instead of 
```
  private boolean isValidKafkaTopic(String key) {
    try {
      Topic.validate(cConf.get(key));
    } catch (InvalidTopicException e) {
      LOG.error("  {} key has invalid kafka topic name. {}", key, e.getMessage());
      return false;
    }
    return true;
  }
```
we could rewrite it to 
```
  private boolean isValidKafkaTopic(String key) {
    try {
      Topic.validate(cConf.get(key));
    } catch (Exception e) {
      LOG.error("  {} key has invalid kafka topic name. {}: {}", key, e.getClass.getSimpleName(), e.getMessage());
      return false;
    }
    return true;
  }
```